agent/dockerclient/dockerapi/docker_client.go

```diff
@@ -230,30 +236,49 @@ type dockerGoClient struct {
 
 func (dg *dockerGoClient) WithVersion(version dockerclient.DockerVersion) DockerClient {
 	return &dockerGoClient{
-		clientFactory: dg.clientFactory,
-		version:       version,
-		auth:          dg.auth,
-		config:        dg.config,
+		clientFactory:    dg.clientFactory,
+		sdkClientFactory: dg.sdkClientFactory,
+		version:          version,
+		auth:             dg.auth,
+		config:           dg.config,
 	}
 }
 
 // scratchCreateLock guards against multiple 'scratch' image creations at once
 var scratchCreateLock sync.Mutex
 
 // NewDockerGoClient creates a new DockerGoClient
-func NewDockerGoClient(clientFactory clientfactory.Factory, cfg *config.Config) (DockerClient, error) {
+// TODO Remove clientfactory parameter once migration to Docker SDK is complete.
+func NewDockerGoClient(clientFactory clientfactory.Factory, sdkclientFactory sdkclientfactory.Factory,
+	cfg *config.Config) (DockerClient, error) {
+	// Ensure both clients can connect to the Docker daemon.
 	client, err := clientFactory.GetDefaultClient()
 
 	if err != nil {
-		seelog.Errorf("DockerGoClient: unable to connect to Docker daemon. Ensure Docker is running: %v", err)
+		seelog.Errorf("DockerGoClient: go-dockerclient unable to connect to Docker daemon. " +
+			"Ensure Docker is running: %v", err)
 		return nil, err
 	}
+	sdkclient, err := sdkclientFactory.GetDefaultClient()
 
-	// Even if we have a dockerclient, the daemon might not be running. Ping it
+	if err != nil {
+		seelog.Errorf("DockerGoClient: Docker SDK client unable to connect to Docker daemon. " +
+			"Ensure Docker is running: %v", err)
+		return nil, err
+	}
+
+	// Even if we have a DockerClient, the daemon might not be running. Ping from both clients
 	// to ensure it's up.
 	err = client.Ping()
 	if err != nil {
-		seelog.Errorf("DockerGoClient: unable to ping Docker daemon. Ensure Docker is running: %v", err)
+		seelog.Errorf("DockerGoClient: go-dockerclient unable to ping Docker daemon. " +
+			"Ensure Docker is running: %v", err)
+		return nil, err
+	}
+	_, err = sdkclient.Ping(context.TODO())
```

Comment:
This should probably not be `context.TODO()`. We should be passing in context from parent.